rclcpp/include/rclcpp/node_interfaces/node_graph.hpp

```diff
@@ -117,9 +118,35 @@ class NodeGraph : public NodeGraphInterface
   size_t
   count_graph_users() override;
 
+  RCLCPP_PUBLIC
+  std::vector<rclcpp::TopicInfo>
+  get_publishers_info_by_topic(
+    const std::string & topic_name,
+    bool no_mangle = false) const override;
+
+  RCLCPP_PUBLIC
+  std::vector<rclcpp::TopicInfo>
+  get_subscriptions_info_by_topic(
+    const std::string & topic_name,
+    bool no_mangle = false) const override;
+
 private:
   RCLCPP_DISABLE_COPY(NodeGraph)
 
+  typedef rcl_ret_t (* rcl_get_info_by_topic_func_t)(
```

Comment:
Yeah, or a template parameter can be used to specify the type, instead of using `std::function`.
note: `std::function` is not a "zero cost" abstraction of a function pointer, I prefer limiting its usage as a container of "any function".

Reply:
Yeah, the type erasure in `std::function` has overhead, but I didn't think it was that much. Using a template parameter would be good, too. Just something more idiomatic in C++ than a function pointer.